crates/ra_assists/src/assists/apply_demorgan.rs

```diff
@@ -0,0 +1,102 @@
+//! This contains the functions associated with the demorgan assist.
+//! This assist transforms boolean expressions of the form `!a || !b` into
+//! `!(a && b)`.
+use hir::db::HirDatabase;
+use ra_syntax::ast::{AstNode, BinExpr, BinOp, Expr, PrefixOp};
```

Comment:
ast type should be used qualified `use ra_syntax::ast::{self, AstNode}`, and then `ast::BinExpr, ast::Expr`